ocs_ci/ocs/amq.py

```diff
@@ -80,77 +90,122 @@ def _clone_amq(self):
     def setup_amq_cluster_operator(self):
         """
         Function to setup amq-cluster_operator,
-        the file file is pulling from github
+        the file is pulling from github
         it will make sure cluster-operator pod is running
         """
 
-        # self.amq_dir = constants.TEMPLATE_DEPLOYMENT_AMQ_CP
+        # Create strimzi-cluster-operator pod
         run(f'oc apply -f {self.amq_dir} -n {self.namespace}', shell=True, check=True, cwd=self.dir)
-        time.sleep(5)
-        # Wait for strimzi-cluster-operator pod to be created
-        if self.is_amq_pod_running(pod_pattern="cluster-operator"):
-            log.info("strimzi-cluster-operator pod is in running state")
-        else:
-            raise ResourceWrongStatusException("strimzi-cluster-operator pod is not getting to running state")
+        time.sleep(10)
 
-        run(f'oc apply -f {self.amq_dir_examples} -n {self.namespace}', shell=True, check=True, cwd=self.dir)
-        # checking pod status one more time
+        #  Check strimzi-cluster-operator pod created
         if self.is_amq_pod_running(pod_pattern="cluster-operator"):
             log.info("strimzi-cluster-operator pod is in running state")
         else:
             raise ResourceWrongStatusException("strimzi-cluster-operator pod is not getting to running state")
 
-    def is_amq_pod_running(self, pod_pattern="cluster-operator"):
+    def is_amq_pod_running(self, pod_pattern):
         """
         The function checks if provided pod_pattern finds a pod and if the status is running or not
+
         Args:
             pod_pattern (str): the pattern for pod
+
         Returns:
             bool: status of pod: True if found pod is running
         """
+
+        _rc = True
+
         for pod in TimeoutSampler(
             300, 10, get_pod_name_by_pattern, pod_pattern, self.namespace
         ):
             try:
-                if pod[0] is not None:
-                    amq_pod = pod[0]
+                if pod is not None:
```

Comment:
I think it will be better to even confirm that we have found the expected number of pods. 
len(pattern_match pods) == expected_pods

WDYT?

Reply:
Agreed, will change accordingly